external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -167,14 +167,12 @@ class AvroDeserializer(rootAvroType: Schema, rootCatalystType: DataType) {
       case (ARRAY, ArrayType(elementType, containsNull)) =>
         val elementWriter = newWriter(avroType.getElementType, elementType, path)
         (updater, ordinal, value) =>
-          val array = value.asInstanceOf[GenericData.Array[Any]]
+          val array = value.asInstanceOf[java.util.Collection[Any]]
           val len = array.size()
           val result = createArrayData(elementType, len)
           val elementUpdater = new ArrayDataUpdater(result)
 
-          var i = 0
-          while (i < len) {
-            val element = array.get(i)
+          for ((element, i) <- array.asScala.zipWithIndex) {
```

Comment:
Why did we change this? `zipWithIndex` is discouraged especially in a performance sensitive code path (https://github.com/databricks/scala-style-guide#perf-whileloops)

Reply:
I kept the old counter.